src/de/jost_net/JVerein/gui/dialogs/BuchungenMitgliedskontenZuordnungDialog.java

```diff
@@ -0,0 +1,528 @@
+          {
+            GUI.getStatusBar().setErrorText("Es wurden keine eindeutigen Mitglieder zum Zuordnen in den gew�hlten Zeitraum gefunden.");
+            return;
+          }
+
+          DBIterator<Buchung> buchungen = Einstellungen.getDBService().createList(Buchung.class);
+          buchungen.addFilter("datum >= ?", dateFromInput);
+          buchungen.addFilter("datum <= ?", dateUntilInput);
+          buchungen.addFilter("mitgliedskonto is null");
+          buchungen.setOrder("ORDER BY datum");
+
+          List<BookingMemberAccountEntry> assignedBooking = new ArrayList<>();
+          Set<String> usedMemberAccount = new HashSet<>();
+
+          while(buchungen.hasNext()) 
+          {
+            Buchung buchung = buchungen.next();
+
+            String bookingPurpose = getBookingPurpose(buchung);
+            if(assginMemberAccountToBooking(assignedBooking,  usedMemberAccount, dateFromInput, dateUntilInput, buchung, uniqueIbans.get(buchung.getIban()), "IBAN") ||
+                assginMemberAccountToBooking(assignedBooking, usedMemberAccount, dateFromInput, dateUntilInput, buchung, uniqueIds.get(bookingPurpose), externeMitgliedsnummer ? "Ext. Mitgliedsnummer" : "Mitgliedsnummer") ||
+                assginMemberAccountToBooking(assignedBooking, usedMemberAccount, dateFromInput, dateUntilInput, buchung, uniqueNames.get(bookingPurpose), "Vorname und Nachname")) 
+            {
+              continue;
+            }
+          }
+
+          if(assignedBooking.isEmpty())
+          {
+            GUI.getStatusBar().setErrorText("Es wurden keine passenden Buchungen oder Mitgliedskonten zum Zuordnen gefunden.");
+          }
+          else
+          {
+            BuchungenMitgliedskontenZuordnungVorschauDialog userValidationDialog = new BuchungenMitgliedskontenZuordnungVorschauDialog(assignedBooking);
+            userValidationDialog.open();
+          }
+        } 
+        catch (RemoteException e)
+        {
+          Logger.error("error while saving import file", e);
+          throw new ApplicationException("Fehler bei der Zuordnungssuche", e);
+        } 
+        catch (Exception e)
+        {
+          Logger.error("error while opening a dialog", e);
+          throw new ApplicationException("Fehler bei der Durchf�hrung der Zuordnung (Best�tigungsdialog)", e);
+        }
+      }
+
+      @Override
+      public void interrupt()
+      {
+        //
+      }
+
+      @Override
+      public boolean isInterrupted()
+      {
+        return false;
+      }
+    };
+
+    Application.getController().start(t);
+
+    close();
+  }
+
+  private String getBookingPurpose(Buchung buchung) throws RemoteException {
+    String zweck = buchung.getZweck();
+    if(zweck == null) return null;
+    zweck = zweck.replaceAll("\r\n", " ").replaceAll("\r", " ").replaceAll("\n", " ").toUpperCase();
+    // manche Banken haengen hier noch zusaetzliche Felder ran: diese versuchen wir Abzuschneiden
+    if(zweck.contains(" IBAN:")) {
```

Comment:
Diese 3 Zeilen können weg, da die IBAN aus dem Verwendungszweck nie verwendet wird.

Reply:
Ich nehme an, dass es um eine, über Hibiscus/XML, eingespielte Buchung geht in der im Verwendungszweck sowas wie "VORNAME NACHNAME IBAN: DE12345 DATUM: XX.YY.ZZZZ" steht und durch das beschneiden dann nur "VORNAME NACHNAME" weiterhin genutzt wird für bookingPurpose.

Reply:
Stimmt.